test/e2e/e2e_test.go

```diff
@@ -521,6 +521,21 @@ func pollForCARotation(t *testing.T, client *kubernetes.Clientset, caCertPEM, ca
 	return secret
 }
 
+// pollForCARecreation polls for the signing secret to be re-created in
+// response to CA secret deletion.
+func pollForCARecreation(client *kubernetes.Clientset) error {
+	return wait.PollImmediate(time.Second, rotationPollTimeout, func() (bool, error) {
+		_, err := client.CoreV1().Secrets(serviceCAControllerNamespace).Get(context.TODO(), signingKeySecretName, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			return false, nil
+		}
+		if err != nil {
+			return false, err
```

Comment:
(no action required) I usually prefer
```
if err != nil {
    if errors.IsNotFound(err) {
        return false, nil
    }
    return false, err
}
```
but I see that that's not the pattern used in this file either :neutral_face: